### ISSUE TITLE
UnCLIPPipeline not supported on Mac/MPS

### ISSUE THREAD

**Issue by Birch-san** [huggingface/diffusers#1791]
### Describe the bug

_Tested on latest diffusers commit, [`7eee6c3`](https://github.com/huggingface/diffusers/tree/4125756e88e82370c197fecf28e9f0b4d7eee6c3)._

When I run UnCLIPPipeline on MPS backend: Python crashes with this error

```
Assertion failed: (vals.isSplat()), function getConstantValue, file CommonRuntimeUtils.h, line 58.
```

I can get past the crash with some MPS-specific fixes:  
https://github.com/Birch-san/diffusers/commit/3efe0138368c4b5c6b5fda572352a1c8bd22d210

this outputs an image, but it's incoherent.  
![00013](https://user-images.githubusercontent.com/6141784/208796071-ba0eb6c2-d008-4063-80b3-04fc9a216ee9.png)  
so probably there was a problem along the way. usually on MPS the problem will be with strided tensors or tensor indexing. or things becoming 0 or ±Inf or NaN.

I'm using PyTorch 1.14.0.dev20221105, on macOS Ventura 13.1 Public Beta 4.

I can't try latest nightly because cumsum regresses, which breaks stable-diffusion:  
https://github.com/pytorch/pytorch/issues/89784

### Reproduction

Use UnCLIPPipeline on an M1 Mac. If you want to try same pytorch nightly as me, you could do this:

```bash
pip install --upgrade --pre torch==1.14.0.dev20221105 --extra-index-url https://download.pytorch.org/whl/nightly/cpu
```

```python
from os import makedirs, listdir, path
import torch
from diffusers import UnCLIPPipeline

pipe = UnCLIPPipeline.from_pretrained("kakaobrain/karlo-v1-alpha", torch_dtype=torch.float16)
pipe = pipe.to("mps")

prompt = "a high-resolution photograph of a big red frog on a green leaf."
image = pipe(prompt).images[0]

sample_path='out'
for path_ in [sample_path]:
  makedirs(path_, exist_ok=True)

base_count = len(listdir(sample_path))
image.save(path.join(sample_path, f"{base_count:05}.png"))
```

### Logs

```shell
NOTE: Redirects are currently not supported in Windows or MacOs.
Fetching 20 files: 100%|████████████████████████████████████████| 20/20 [00:00<00:00, 6038.88it/s]
/Users/birch/git/diffusers-play/src/diffusers/src/diffusers/pipelines/unclip/pipeline_unclip.py:144: UserWarning: The operator 'aten::repeat_interleave.self_int' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1667632147621/work/aten/src/ATen/mps/MPSFallback.mm:11.)
  text_embeddings = text_embeddings.repeat_interleave(num_images_per_prompt, dim=0)
100%|█████████████████████████████████████████████████████████████| 25/25 [00:02<00:00,  9.88it/s]
100%|█████████████████████████████████████████████████████████████| 25/25 [00:09<00:00,  2.53it/s]
100%|███████████████████████████████████████████████████████████████| 7/7 [00:06<00:00,  1.04it/s]
```
```


### System Info

```
- `diffusers` version: 0.8.0.dev0
- Platform: macOS-13.1-arm64-64bit
- Python version: 3.10.6
- PyTorch version (GPU?): 1.14.0.dev20221105 (False)
- Huggingface_hub version: 0.10.1
- Transformers version: 4.25.1
- Using GPU in script?: MPS
- Using distributed or parallel set-up in script?: no
```

**Comment by pcuenca** [comment]
Thanks for reporting @Birch-san, I'll take a look!

**Comment by patrickvonplaten** [comment]
Also cc @williamberman here

**Comment by williamberman** [comment]
As part of https://github.com/huggingface/diffusers/pull/1908 I'll temporarily disable mps tests for the unclip pipeline

**Comment by patrickvonplaten** [comment]
@williamberman do the MPS tests also fail?

**Comment by williamberman** [comment]
> @williamberman do the MPS tests also fail? 

Correct! With the same assertion error posted here

**Comment by williamberman** [comment]
https://github.com/huggingface/diffusers/pull/1908 I added fixes in the locations @Birch-san pointed out in this PR. So when merged we should have the standard chunk of the test suite that we run against MPS running for unCLIP but will still have the bad outputs :)

**Comment by pcuenca** [comment]
Update after testing this today:
- `fp16` crashed. If the crash is fixed (#2258), the output image is black.
- Using `float32` the output image is still malformed as @Birch-san pointed out.

This requires a step-by-step investigation to see where the problem happens, not sure if I'll be able to have time in the next few days.

**Comment by patrickvonplaten** [comment]
I don't think it's that urgent to have MPS for UnCLIP to be honest. Think we can see if the community figures out a solution here or not :-)
